core-libs/setup/ssr/optimized-engine/ssr-optimization-options.ts

```diff
@@ -164,6 +167,7 @@ export const defaultSsrOptimizationOptions: SsrOptimizationOptions = {
     defaultRenderingStrategyResolverOptions
   ),
   logger: new DefaultExpressServerLogger(),
-  cachingStrategyResolver: (entry) => !entry.error,
-  ssrErrorHandling: true,
+  cacheStrategyResolver: (options, entry) =>
+    !!options.cacheErrors && !!entry.error,
```

Comment:
1. we have to check if  `cacheErrors` is expliclity `=== true`. Becuase otherwise also `undefined` makes it falsy by default - and we dont want to toggle it falsy by default.
2. IMHO semantics of a flag should be **falsy** when disabled (by default) and **truthy** when enabled. 

What about switching the meaning to `avoidCachingErrors`. IMHO it's more attactive and understandable for customers to toggle it on